adafruit_tlc59711/adafruit_tlc59711_multi.py

```diff
@@ -0,0 +1,1120 @@
+        :return tuple: Ioclmax (mA)
+        """
+        # Riref                 = (Viref / Ioclmax) * 41    | / 41
+        # Riref / 41            = Viref / Ioclmax           | switch
+        # 41 / Riref            = Ioclmax / Viref           | * Viref
+        # (41 / Riref) * Viref  = Ioclmax
+        if not 0.8 <= Riref <= 24.8:
+            raise ValueError(
+                "Riref {} not in range: 0.8kΩ..25kΩ"
+                "".format(Riref)
+            )
+        Viref = 1.21
+        Ioclmax = (41 / Riref) * Viref
+        if not 2.0 <= Ioclmax <= 60.0:
+            raise ValueError(
+                "Ioclmax {} not in range: 2mA..60mA"
+                "".format(Ioclmax)
+            )
+        return Ioclmax
+
+    @staticmethod
+    def calculate_Riref(
+            *,
+            Ioclmax=20,
+    ):
+        """
+        Calculate Maximum Constant Sink Current Value.
+
+        see:
+        8.4.1 Maximum Constant Sink Current Setting
+        http://www.ti.com/lit/ds/symlink/tlc5971.pdf#page=19&zoom=200,0,697
+
+        Riref = (Viref / Ioclmax) * 41
+
+        :param float Ioclmax: target max output current (mA) (default=20)
+        :return tuple: Riref (kΩ)
+        """
+        if not 2.0 <= Ioclmax <= 60.0:
+            raise ValueError(
+                "Ioclmax {} not in range: 2mA..60mA"
+                "".format(Ioclmax)
+            )
+        Viref = 1.21
+        Riref = (Viref / Ioclmax) * 41
+        if not 0.8 <= Riref <= 24.8:
+            raise ValueError(
+                "Riref {} not in range: 0.8kΩ..25kΩ"
+                "".format(Riref)
+            )
+        return Riref
+
+    @staticmethod
+    def calculate_BCData(
+            *,
+            Ioclmax=18,
+            IoutR=17,
+            IoutG=15,
+            IoutB=9
+    ):
+        """
+        Calculate Global Brightness Control Values.
+
+        see:
+        8.5.1 Global Brightness Control (BC) Function (Sink Current Control)
+        http://www.ti.com/lit/ds/symlink/tlc5971.pdf#page=19&zoom=200,0,697
+
+        Iout = Ioclmax * (BCX / 127)
+        BCX = Iout / Ioclmax * 127
+
+        :param float Ioclmax: max output current set by Riref (mA) (default=20)
+        :param float IoutR: max output current for red color group (mA)
+            (default=9)
+        :param float IoutG: max output current for green color (mA)
+            (default=15)
+        :param float IoutB: max output current for blue color (mA)
+            (default=17)
+        :return tuple: (bcr, bcg, bcb)
+        """
+        # Iout                 =  Ioclmax * (BCX / 127)  | / Ioclmax
+        # Iout / Ioclmax       =  BCX / 127              | * 127
+        # Iout / Ioclmax * 127 =  BCX
+        if not 2.0 <= Ioclmax <= 60.0:
+            raise ValueError(
+                "Ioclmax {} not in range: 2mA..60mA"
+                "".format(Ioclmax)
+            )
+        if not 0.0 <= IoutR <= Ioclmax:
+            raise ValueError(
```

Comment:
Put on one or two lines, and below.

Reply:
what is your prefered style with this:
my original
```
        if not 2.0 <= Ioclmax <= 60.0:
            raise ValueError(
                "Ioclmax {} not in range: 2mA..60mA"
                "".format(Ioclmax)
            )
```
this is all beautifully aligned in 4spaces..
if i understand you correctly:
```
        if not 2.0 <= Ioclmax <= 60.0:
            raise ValueError("Ioclmax {} not in range: 2mA..60mA"
                "".format(Ioclmax))
```
with this my pylama (pep8) errors with
`E128 continuation line under-indented for visual indent`
so i would need to fi this so:
```
        if not 2.0 <= Ioclmax <= 60.0:
            raise ValueError("Ioclmax {} not in range: 2mA..60mA"
                             "".format(Ioclmax))
```
but this means that the last line is not in multiples of 4 anymore..

please let me know which way i should correct the code :-)
then i will do my best to fix all occurrences..

Reply:
have decided to a compromise ;-)
hope that is fine with you. (and i hope i have fixed all occurrences.)